Apps/Contoso.Forms.Demo/Contoso.Forms.Demo.Droid/MainActivity.cs

```diff
@@ -44,11 +44,11 @@ protected override void OnActivityResult(int requestCode, Result resultCode, Int
             if (requestCode == FileAttachmentId)
             {
                 var uri = resultCode == Result.Ok && data != null ? data.Data : null;
-                if (Build.VERSION.SdkInt >= BuildVersionCodes.Kitkat && uri != null)
+                if (uri != null)
                 {
                     ContentResolver.TakePersistableUriPermission(uri, data.Flags & ActivityFlags.GrantReadUriPermission);
+                    FileAttachmentTaskCompletionSource.SetResult(uri?.ToString());
```

Comment:
I guess you should set the result even if url is null. or remove `?.` because it's already checked before